pkgs/native_assets_builder/test/build_runner/helpers.dart

```diff
@@ -52,6 +52,7 @@ Future<BuildResult> build(
       final result = await NativeAssetsBuildRunner(
         logger: logger,
         dartExecutable: dartExecutable,
+        progressLogger: logger.info,
```

Comment:
This doesn't replace the previous progress with the new one right? It just prints all progress messages.

Reply:
Yes!

Reply:
I think we should have at least one test that exercises the replacing of messages in a terminal or something.
Ditto for the counting and updating seconds that Flutter has for the other build steps. (No idea if that should be implemented in this package or in flutter_tools and dartdev. But if it's in this package we don't have to duplicate it.)

Reply:
What most of the build systems do in the terminal. Replace the last line:
<img width="355" alt="image" src="https://github.com/user-attachments/assets/5a8ff718-3a7c-4c52-be41-693d270112e4">

Reply:
Otherwise I don't see why we'd need a separate logger?